File: api/src/endpoint/user/list_user_repositories.rs
use serde::{Deserialize, Serialize};
use api_derive::EndpointRequest;

use crate::{endpoint::Endpoint, resource::repository::RepositoryResource};

pub struct ListUserRepositories;

impl Endpoint for ListUserRepositories {
    const PATH: &'static str = "/user/{user_name}/repositories";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListUserRepositoriesRequest;
    type Response = ListUserRepositoriesResponse;
}

#[derive(EndpointRequest, Debug, Serialize, Deserialize)]
pub struct ListUserRepositoriesRequest {}

pub type ListUserRepositoriesResponse = Vec<RepositoryResource>;

File: gitdot-api-ts/src/endpoint/user/list-user-repositories.ts
import { z } from "zod";

import { RepositoryResource } from "../../resource";

export const ListUserRepositoriesRequest = z.object({});
export type ListUserRepositoriesRequest = z.infer<
  typeof ListUserRepositoriesRequest
>;

export const ListUserRepositories = {
  path: "/user/{user_name}/repositories",
  method: "GET",
  request: ListUserRepositoriesRequest,
  response: z.array(RepositoryResource),
} as const;
export type ListUserRepositories = typeof ListUserRepositories;
